### PR TITLE
Speedup of metric tests

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Partly fixes https://github.com/PyTorchLightning/pytorch-lightning/issues/4059
This PR adds back our metric test.
Locally I experienced a reduction from 476.71sec before to 18.68sec (factor 26!) running the 499 tests currently in metrics.
This is mainly achieved by only starting ddp once per metric (instead of one per parameterized test) and reusing the same pool of processes during testing.

Also added a bit of description here and there to the testing functions

# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
@SkafteNicki #4115 is merged, you just need to remove this like:
https://github.com/PyTorchLightning/pytorch-lightning/blob/master/setup.cfg#L22

**Comment by teddykoker** [comment]
Great work @SkafteNicki thanks for this :). What else is left to do? Would be great if we could get the tests back ASAP

**Comment by teddykoker** [comment]
> @SkafteNicki #4115 is merged, you just need to remove this like:
> https://github.com/PyTorchLightning/pytorch-lightning/blob/master/setup.cfg#L22

Done :)

**Review by teddykoker — APPROVED** [review]

**Comment by SkafteNicki** [comment]
Coverage back over 90%!

**Review by ananyahjha93 — APPROVED** [review]

File: tests/metrics/classification/test_f_beta.py [hunk 1]
```diff
@@ -1,81 +1,76 @@
...
-from pytorch_lightning.metrics import Fbeta
+import numpy as np
+import pytest
+import torch
 from sklearn.metrics import fbeta_score
 
-from tests.metrics.utils import compute_batch, setup_ddp
-from tests.metrics.utils import NUM_BATCHES, NUM_PROCESSES, BATCH_SIZE, NUM_CLASSES, THRESHOLD
-
+from pytorch_lightning.metrics import Fbeta
```
Comment (Borda) [inline tests/metrics/classification/test_f_beta.py:8]: can we have this as `FBeta` to be consistent?

**Review by justusschock — APPROVED** [review]
Great work @SkafteNicki

File: tests/metrics/classification/test_f_beta.py [hunk 2]
```diff
@@ -1,81 +1,76 @@
...
+    _multilabel_inputs,
+    _multilabel_prob_inputs,
 )
+from tests.metrics.utils import BATCH_SIZE, NUM_BATCHES, NUM_CLASSES, NUM_PROCESSES, THRESHOLD, MetricTester
 
 torch.manual_seed(42)
 
 
-def _binary_prob_sk_metric(preds, target, average='micro', beta=1.):
+def _binary_prob_sk_metric(preds, target, average='micro', beta=1.0):
```
Comment (Borda) [inline tests/metrics/classification/test_f_beta.py:24]: can we rename these as `_sk_accuracy_binary_prob` so it clear which SK is used... like this, you fined multiple equal named fiction in several files and it very confusing...

File: tests/metrics/classification/test_f_beta.py (hunk 1 above)
Comment (teddykoker) [reply tests/metrics/classification/test_f_beta.py:8]: @ananyahjha93

Comment (Borda) [reply tests/metrics/classification/test_f_beta.py:8]: if you agree I can integrate it in #4152
